tests/test_signals.py

```diff
@@ -20,12 +21,27 @@ async def test_receive_signals():
 
 
 @pytest.mark.skipif(sys.platform == 'win32', reason='Signal delivery cannot be tested on Windows')
-async def test_task_group_cancellation():
+async def test_task_group_cancellation_open() -> None:
     async def signal_handler():
         async with open_signal_receiver(signal.SIGUSR1) as sigiter:
             async for v in sigiter:
                 assert False
+            assert False
+        assert False
 
     async with create_task_group() as tg:
         await tg.spawn(signal_handler)
         await tg.cancel_scope.cancel()
+
+
+@pytest.mark.skipif(sys.platform == 'win32', reason='Signal delivery cannot be tested on Windows')
```

Comment:
These could be move to the `pytestmark` collection since there are more than one now.